### PR TITLE
Query log extraction for Postgres crawler

### PR DESCRIPTION
**Pull request by elic-eon**
### 🤔 Why?

PostgreSQL does not store the query history, or user action in the database but can be configured as logging to its log file. Therefore, we can parse the Postgres log for query log extraction

### 🤓 What?

- Extract query history from CloudWatch

### 🧪 Tested?

Test against an RDS instance.

### ☑️ Checks

- [x] My PR contains actual code changes, and I have updated the version number in `pyproject.toml`.

### CODE REVIEW

File: metaphor/postgresql/README.md [hunk 1]
```diff
@@ -41,6 +41,30 @@ By default, the connector will connect using the default PostgreSQL port 5432. Y
 port: <port_number>
 ```
 
+### Query Log Extraction Configurations
+
+The connector support extract query log history from CloudWatch for RDS postgres. Please follow the [documentation](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/USER_LogAccess.Concepts.PostgreSQL.Query_Logging.html) to configure RDS to log SQL statement to CloudWatch.
```
Comment (mars-lan) [inline metaphor/postgresql/README.md:46]:
```suggestion
The connector supports extracting query log history from CloudWatch for RDS PostgreSQL. Please follow [this documentation](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/USER_LogAccess.Concepts.PostgreSQL.Query_Logging.html) to configure RDS to log SQL statements to CloudWatch.
```

File: metaphor/postgresql/README.md [hunk 2]
```diff
@@ -41,6 +41,30 @@ By default, the connector will connect using the default PostgreSQL port 5432. Y
...
+
+```yaml
+query_log:
+  aws:
+    access_key_id: <aws_access_key_id>
+    secret_access_key: <aws_secret_access_key>
+    region_name: <aws_region_name>
+    assume_role_arn: <aws_role_arn>  # If using IAM role
+    session_token: <aws_session_token>  # If using session token
+    profile_name: <aws_profile_name>  # If using AWS profile
```
Comment (mars-lan) [inline metaphor/postgresql/README.md:56]: I think we drop `session_token` & `profile_name`. While the SDK takes those parameters, setting them in crawlers is rather uncommon, especially for fully managed ones.

File: metaphor/redshift/config.py [hunk 3]
```diff
@@ -27,4 +27,4 @@ class RedshiftRunConfig(PostgreSQLRunConfig):
     # configs for fetching query logs
     query_log: RedshiftQueryLogConfig = field(
         default_factory=lambda: RedshiftQueryLogConfig()
-    )
+    ) # type: ignore
```
Comment (mars-lan) [inline metaphor/redshift/config.py:30]: Why?

Comment (elic-eon) [reply metaphor/redshift/config.py:30]:
```
metaphor/redshift/config.py:28: error: Incompatible types in assignment (expression has type "RedshiftQueryLogConfig", base class "PostgreSQLRunConfig" defined the type as "PostgreSQLQueryLogConfig")  [assignment]```

mypy think it's wrong

Comment (mars-lan) [reply metaphor/redshift/config.py:30]: Maybe it's time to adopt [composition over inheritance](https://en.wikipedia.org/wiki/Composition_over_inheritance)

Comment (elic-eon) [reply metaphor/redshift/config.py]: SG

File: tests/postgresql/test_extractor.py [hunk 4]
```diff
@@ -40,3 +45,121 @@ def test_parse_format_type():
...
+            cache,
+        )
+        is None
+    )
+    assert (
+        extractor._process_cloud_watch_log(
+            "2024-08-29 09:25:51 UTC:10.1.1.134(48507):metaphor@metaphor:[615]:LOG: duration: 55.66 ms",
+            cache,
+        )
+        is None
```
Comment (usefulalgorithm) [inline tests/postgresql/test_extractor.py:134]: why does this not create a query log?

File: tests/postgresql/test_extractor.py [hunk 5]
```diff
@@ -40,3 +45,121 @@ def test_parse_format_type():
...
```
Comment (usefulalgorithm) [inline tests/postgresql/test_extractor.py:149]: and this? this looks identical to the one that generates a query log above

Comment (elic-eon) [reply tests/postgresql/test_extractor.py:154]: Added comment in test case, `foo` is in the `exclude_usernames`

File: tests/postgresql/test_extractor.py (hunk 4 above)
Comment (elic-eon) [reply tests/postgresql/test_extractor.py:138]: We don't want to collect queries without connection to any table

**Review by usefulalgorithm — APPROVED** [review]

**Review by usefulalgorithm — CHANGES_REQUESTED** [review]
LGTM, just gotta fix mypy

**Review by usefulalgorithm — APPROVED** [review]

**Pull request merged**